snapcraft/tests/test_lifecycle.py

```diff
@@ -618,6 +618,31 @@ def test_pull_is_dirty_if_target_arch_changes(
             "by running: snapcraft clean part1 -s pull\n",
             str(raised))
 
+    def test_prime_records_snapcraft_yaml(self):
+        self.make_snapcraft_yaml("""parts:
+  test-part:
+    plugin: nil
+""")
+        lifecycle.execute('prime', self.project_options)
+
+        expected = ("""name: test
+version: 0
+summary: test
+description: test
+confinement: strict
+grade: stable
+parts:
+  test-part:
+    plugin: nil
+    prime: []
```

Comment:
The default values for prime and snap are not [], they are ['*']. We should take them too from the state files, right?